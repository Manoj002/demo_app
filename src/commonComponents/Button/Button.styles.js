import styled, {css} from 'styled-components/native';
import {Animated} from 'react-native';
import {Colors} from '../../constants';
import {Spacing} from '../../constants';

export const ButtonWrapper = styled.TouchableOpacity`
  width: ${Spacing.SPACE_100}%;
  height: ${Spacing.SPACE_50}px;
  align-self: center;
  border-radius: ${Spacing.SPACE_8}px;
  align-items: center;
  justify-content: center;
  ${({marginTop}) => marginTop && `margin-top:${marginTop}px`};
  ${({marginBottom}) => marginBottom && `margin-bottom:${marginBottom}px`};
  ${({buttonAsText, faded, colored}) => {
    if (buttonAsText) {
      return null;
    }
    if (faded) {
      return css`
        background-color: ${Colors.lightBlue};
      `;
    }
    if (colored) {
      return css`
        background-color: ${Colors.blue};
      `;
    }
  }}
`;

export const SwipeButtonWrapper = styled.View`
  width: ${Spacing.SPACE_100}%;
  height: ${Spacing.SPACE_50}px;
`;

export const AnimatedButtonWrapper = styled.View`
  width: ${Spacing.SPACE_100}%;
  height: ${Spacing.SPACE_50}px;
  flex-direction: row;
  align-self: center;
  border-radius: ${Spacing.SPACE_8}px;
  border-width: ${Spacing.SPACE_2}px;
  border-color: ${Colors.lightWhite};
  align-items: center;
  justify-content: center;
  ${({marginTop}) => marginTop && `margin-top:${marginTop}px`};
  ${({marginBottom}) => marginBottom && `margin-bottom:${marginBottom}px`};
`;

export const ScrollableView = styled(Animated.View)`
  align-items: center;
  justify-content: center;
  position: absolute;
  left: -2px;
  top: -2px;
  z-index: 1;
  width: ${Spacing.SPACE_50}px;
  height: ${Spacing.SPACE_50}px;
  border-radius: ${Spacing.SPACE_8}px;
  background-color: ${Colors.blue};
`;
