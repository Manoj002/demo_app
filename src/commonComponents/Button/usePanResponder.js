import {useReducer, useRef} from 'react';
import {PanResponder, useWindowDimensions} from 'react-native';

import {actionCreators, initialState, reducer} from './pan';

export default function usePanResponder() {
  const [state, dispatch] = useReducer(reducer, initialState);

  const windowWidth = useWindowDimensions().width;

  const handlePanResponderMove = (e, gestureState) => {
    if (
      gestureState.dx > 0 &&
      gestureState.dx + state.initialX < windowWidth - 113
    ) {
      dispatch(actionCreators.move({x: gestureState.dx, y: 0}));
    }
  };

  const handlePanResponderEnd = (e, gestureState) => {
    if (
      gestureState.dx > 0 &&
      gestureState.dx + state.initialX < windowWidth - 113
    ) {
      dispatch(actionCreators.end({x: gestureState.dx, y: 0}));
    }
  };

  const panResponder = useRef(
    PanResponder.create({
      onStartShouldSetPanResponder: () => true,
      onPanResponderMove: handlePanResponderMove,
      onPanResponderRelease: handlePanResponderEnd,
    }),
  );

  return [state, panResponder.current.panHandlers];
}
