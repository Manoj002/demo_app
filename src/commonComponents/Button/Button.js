/* eslint-disable react/jsx-no-undef */
import React from 'react';
import {StyleSheet} from 'react-native';
import usePanResponder from './usePanResponder';
import Text from '../Text';
import PropTypes from 'prop-types';
// import SwipeButton from 'rn-swipe-button';
import Icon from 'react-native-vector-icons/FontAwesome';
// import Strings from '../../constants/Strings';
import {
  ButtonWrapper,
  // SwipeButtonWrapper,
  AnimatedButtonWrapper,
  ScrollableView,
} from './Button.styles';

const Button = ({
  children,
  buttonAsText,
  faded,
  colored,
  opacity,
  onPress,
  marginBottom,
  marginTop,
  animated,
  testID,
}) => {
  const [state, panHandlers] = usePanResponder();
  const {initialY, initialX, offsetY, offsetX} = state;

  const getColor = () => {
    if (buttonAsText || faded || animated) {
      return 'blue';
    }
    if (colored) {
      return 'white';
    }
  };

  const style = {
    transform: [
      {translateX: initialX + offsetX},
      {translateY: initialY + offsetY},
    ],
  };

  // const DiamondIcon = () => <Icon name="diamond" color="white" size={20} />;

  return (
    <If condition={animated}>
      <AnimatedButtonWrapper
        onPress={onPress}
        marginBottom={marginBottom}
        marginTop={marginTop}
        activeOpacity={opacity || 0.6}
        testID={testID}>
        <ScrollableView style={style} {...panHandlers}>
          <Icon name="diamond" color="white" size={20} />
        </ScrollableView>
        <Text color={getColor()}>{children}</Text>
      </AnimatedButtonWrapper>
      {/* <SwipeButtonWrapper>
        <SwipeButton
          onSwipeFail={() => console.log('swipe failed')}
          onSwipeStart={() => console.log('swipe started')}
          onSwipeSuccess={() => console.log('swipe success')}
          testID="slider-button"
          height={50}
          titleFontSize={16}
          thumbIconStyles={styles.borderRadius}
          thumbIconComponent={DiamondIcon}
          railBackgroundColor="transparent"
          railBorderColor="#343A42"
          railFillBorderColor="#343A42"
          railFillBackgroundColor="red"
          titleColor="#007aff"
          thumbIconBackgroundColor="#007aff"
          title={Strings.SLIDE_ME_TO_CONTINUE}
          railStyles={styles.sliderBackgroundColorStyle}
          containerStyle={styles.containerStyles}
        />
      </SwipeButtonWrapper> */}
      <Else />
      <ButtonWrapper
        buttonAsText={buttonAsText}
        faded={faded}
        colored={colored}
        onPress={onPress}
        marginBottom={marginBottom}
        marginTop={marginTop}
        activeOpacity={buttonAsText ? 1 : opacity || 0.6}
        testID={testID}>
        <Text color={getColor()}>{children}</Text>
      </ButtonWrapper>
    </If>
  );
};

Button.propTypes = {
  children: PropTypes.string,
  buttonAsText: PropTypes.bool,
  faded: PropTypes.bool,
  colored: PropTypes.bool,
  animated: PropTypes.bool,
  opacity: PropTypes.number,
  marginBottom: PropTypes.number,
  marginTop: PropTypes.number,
};

Button.defaultProps = {
  children: null,
  buttonAsText: false,
  faded: false,
  colored: true,
  animated: false,
  opacity: 0.6,
  marginBottom: 0,
  marginTop: 0,
};

export default Button;

const styles = StyleSheet.create({
  borderRadius: {
    borderRadius: 10,
    borderWidth: 0,
  },
  sliderBackgroundColorStyle: {
    backgroundColor: 'transparent',
    borderRadius: 0,
  },
  containerStyles: {
    backgroundColor: 'red',
    borderRadius: 0,
  },
});
