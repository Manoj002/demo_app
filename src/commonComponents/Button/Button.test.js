import React from 'react';
import {fireEvent, render} from '@testing-library/react-native';
import Button from './Button';

describe('Button Component', () => {
  it('should match snapShot', () => {
    expect(render(<Button buttonAsText>Press me</Button>)).toMatchSnapshot();
    expect(render(<Button faded>Press me</Button>)).toMatchSnapshot();
    expect(render(<Button colored>Press me</Button>)).toMatchSnapshot();
    expect(render(<Button opacity={0.8}>Press me</Button>)).toMatchSnapshot();
    expect(
      render(<Button marginBottom={20}>Press me</Button>),
    ).toMatchSnapshot();
    expect(render(<Button marginTop={20}>Press me</Button>)).toMatchSnapshot();
    expect(render(<Button animated>Press me</Button>)).toMatchSnapshot();
  });

  it('should call onPress when button is pressed', () => {
    const handleOnPress = jest.fn();
    const {getByTestId} = render(
      <Button colored onPress={handleOnPress} testID="Press-me-btn">
        Press me
      </Button>,
    );
    const PressMeButton = getByTestId('Press-me-btn');
    fireEvent.press(PressMeButton);
    expect(handleOnPress).toBeCalledTimes(1);
  });
});
