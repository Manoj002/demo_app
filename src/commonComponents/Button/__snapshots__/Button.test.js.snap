// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapShot 1`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="blue"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 2`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#334249",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="blue"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 3`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="white"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 4`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="white"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 5`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "marginBottom": 20,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="white"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 6`] = `
<View
  accessible={true}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "backgroundColor": "#007AFF",
      "borderRadius": 8,
      "height": 50,
      "justifyContent": "center",
      "marginTop": 20,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    align="left"
    bold={false}
    color="white"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;

exports[`Button Component should match snapShot 7`] = `
<View
  activeOpacity={0.6}
  marginBottom={0}
  marginTop={0}
  style={
    Array [
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "borderColor": "#343A42",
        "borderRadius": 8,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "#007AFF",
        "borderRadius": 8,
        "height": 50,
        "left": 0,
        "position": "absolute",
        "top": -2,
        "transform": Array [
          Object {
            "translateX": 0,
          },
          Object {
            "translateY": 0,
          },
        ],
        "width": 50,
      }
    }
  />
  <Text
    align="left"
    bold={false}
    color="blue"
    fontSize={16}
    lineHeight={18}
    marginBottom={0}
    marginLeft={0}
    marginTop={0}
    style={
      Array [
        Object {
          "color": "#007AFF",
          "fontFamily": "Roboto",
          "fontSize": 16,
          "lineHeight": 18,
          "textAlign": "left",
        },
      ]
    }
  >
    Press me
  </Text>
</View>
`;
